edge/pkg/edgestream/edgestream.go

```diff
@@ -42,18 +44,33 @@ type edgestream struct {
 
 var _ core.Module = (*edgestream)(nil)
 
-func newEdgeStream(enable bool, hostnameOverride, nodeIP string) *edgestream {
+func newEdgeStream(enable bool, hostnameOverride, nodeIP string) (*edgestream, error) {
+	var err error
+	if nodeIP == "" {
+		nodeIP, err = util.GetLocalIP(util.GetHostname())
+		if err != nil {
+			klog.Errorf("Failed to get Local IP address: %v", err)
+			return nil, err
+		}
+		klog.Infof("Get node local IP address successfully: %s", nodeIP)
+	}
+
```

Comment:
We can open an issue to track, whether it's need to detect and update the nodeIP periodly in all models which use the nodeIP